neo/Oracle/OracleFilters.cs

```diff
@@ -0,0 +1,125 @@
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Xml;
+using System.Xml.XPath;
+
+namespace Neo.Oracle
+{
+    public class OracleFilters
+    {
+        /// <summary>
+        /// Filter json
+        /// </summary>
+        /// <param name="input">Input</param>
+        /// <param name="filter">Filter</param>
+        /// <param name="output">output</param>
+        /// <returns>True if was filtered</returns>
+        public static bool FilterJson(string input, string filter, out string output)
```

Comment:
Why do we need filter?

Reply:
In order to reduce the size and the agreement.
Imagine this Response

```{"name":"neo":"timestamp":1231239139123}```

Every download you get different values, but the SC only want the "name" property, so the OracleNodes have an easy work

Reply:
SYSCALL should not do too much work, otherwise it will be difficult to price.

We should create an oracle standard that requires a timestamp or id on the Url, and then asks the same request to return the json must be exactly the same. The filtering work should be handed over to the contract.

Reply:
But then all the oracles will be centralized by the projects, because they will need to put a web service for handle the request, with filters we can consume third party services also, like coinmarketcap for example

Reply:
In Oracle case, shouldn't the syscall actually do nothing but take the result from OracleResultCache for CN and regular nodes?

The actual work is done only by Oracle nodes, but they get paid for resulting Oracle Tx, so pricing should not be the problem here. Also, filtering on SC side would increase the OracleTx size and, as a result, increase Oracle usage price and blockchain disk consumption.

Shouldn't Oracle, serving as a way to talk to external world, do as much as possible on Oracle side to decrease the load on SC and blockchain, improve Developer's Experience and provide better support for external world connectivity?

Reply:
It won't be centralized. We shall have a standard. In fact, your PR has asked all the returns to be JSON, which is actually part of the standard I am talking about. We also need some additional specifications. For example, it is required to return the exact same response for the same request.

As for filtering, we have the JSON API, so the contract itself can easily filter JSON.

Reply:
why filtering specifically for oracles? Arent we doing that already as callbacks? I mean, this has general application, not only for oracles.

Reply:
@igormcoelho Filtering for Oracles should be done to save space in blockchain by recording only the filtered result, reduce Oracle Tx size and, indirectly, increase TPS.

Reply:
I mean, its not important just for oracles... its a general purpose approach for filtering data on smart contracts, and if this is the direction, we may need to extend other things as well. I responded on oracle filter issue with more details.

Reply:
Can we move the discussion to the issue? https://github.com/neo-project/neo/issues/1259